src/Microsoft.Health.Fhir.Shared.Api/Features/ActionResults/FhirResult.cs

```diff
@@ -46,6 +48,28 @@ public static FhirResult Create(ResourceElement resource, HttpStatusCode statusC
             };
         }
 
+        /// <summary>
+        /// Creates a FHIR result with the specified parameters
+        /// </summary>
+        /// <param name="bundleResponse">The resonse bundle.</param>
+        public static FhirResult Create(BundleResponse bundleResponse)
+        {
+            EnsureArg.IsNotNull(bundleResponse, nameof(bundleResponse));
+
+            if (bundleResponse.BundleProcessingStatus == BundleProcessingStatus.FAILED)
+            {
+                return new FhirResult(bundleResponse.Bundle)
+                {
+                    StatusCode = HttpStatusCode.BadRequest,
```

Comment:
Bundle could be failed to process for various reasons. Is it always going to be BadRequest (meaning client has send bad request)?

Reply:
I have now parsed the status code from operation outcome and used that to map the proper exception but still defaults to badrequest.